tests/virt_autotest/libvirt_virtual_network_init.pm

```diff
@@ -45,31 +45,30 @@ sub run_test {
     virt_autotest::virtual_network_utils::enable_libvirt_log();
 
     #VM HOST SSH SETUP
-    virt_autotest::virtual_network_utils::ssh_setup();
+    virt_autotest::virtual_network_utils::ssh_setup() if (!(get_var('VIRT_AUTOTEST')));
 
     #Backup file /etc/hosts before virtual network testing
     virt_autotest::virtual_network_utils::hosts_backup();
 
     #Install required packages
     zypper_call '-t in iproute2 iptables iputils bind-utils sshpass nmap';
 
-    #Check with Guest status before libvirt virtual network tests
-    virt_autotest::virtual_network_utils::check_guest_status();
-
     #Prepare Guests
     foreach my $guest (keys %xen::guests) {
         #Archive deployed Guests
         #NOTE: Keep Archive deployed Guests for restore_guests func
         assert_script_run("virsh dumpxml $guest > /tmp/$guest.xml");
         upload_logs "/tmp/$guest.xml";
-        #Start installed Guests
-        assert_script_run("virsh start $guest", 60);
-        #Wait for forceful boot up guests
-        sleep 60;
+        #Used with attach-detach(hotplugging) interface to confirm all virtual network mode
+        #NOTE: Required all guests keep running status
+        #Check that all guests are still running before virtual network tests
+        script_retry("nmap $guest -PN -p ssh | grep open", delay => 30, retry => 6, timeout => 180);
         save_guest_ip($guest, name => "br123");
-        my $mode            = is_sle('=11-sp4') ? '' : '-f';
-        my $default_ssh_key = "/root/.ssh/id_rsa.pub";
-        exec_and_insert_password("ssh-copy-id -i $default_ssh_key -o StrictHostKeyChecking=no $mode root\@$guest");
+        if (!(get_var('VIRT_AUTOTEST'))) {
+            my $mode            = is_sle('=11-sp4') ? '' : '-f';
+            my $default_ssh_key = "/root/.ssh/id_rsa.pub";
+            exec_and_insert_password("ssh-copy-id -i $default_ssh_key -o StrictHostKeyChecking=no $mode root\@$guest");
+        }
```

Comment:
Can you also explain?

Reply:
What can happen if default_ssh_key does not exists ?

Reply:
Refer to the code of setup_dns_service, after created **/var/testvirt.net/.ssh/id_rsa.pub** and modify /etc/ssh/ssh_config with **IdentityFile /var/testvirt.net/.ssh/id_rsa** on vm host, so, there should be still following ssh_config changes, do not care the default ssh_key (i.e. **/root/.ssh/id_rsa**)

Reply:
> Can you also explain?

As my mentioned, After finished setup_dns_service, there was created testing SSH KEY as /var/testvirt.net/.ssh/id_rsa.pub and copy this key to guest system. 
This code changed just only work for QAM environment.

Reply:
I have two questions:
 1) Why don't you modify `virt_autotest::virtual_network_utils::ssh_setup()` to work with your custom SSH key?
 2) Why do you create SSH key in `/var/testvirt.net/.ssh/` and what else is in `/var/testvirt.net/` directory?

Reply:
Where do you copying the SSH key to the guests if not here? If you use another SSH key path, just modify it here:
```suggestion
        my $mode            = is_sle('=11-sp4') ? '' : '-f';
        my $default_ssh_key = (!(get_var('VIRT_AUTOTEST'))) ? "/root/.ssh/id_rsa.pub" : "/var/testvirt.net/.ssh/id_rsa.pub";
        exec_and_insert_password("ssh-copy-id -i $default_ssh_key -o StrictHostKeyChecking=no $mode root\@$guest");
```